### ISSUE TITLE
CsvTest::testGetAPI fails for php 5.4.16

### ISSUE THREAD

**Issue by DieterDePaepe** [tdt/core#340]
I am deploying the datatank on a CentOS 7.1 machine (a clean VirtualBox image). When running the phpunit tests, a single test fails: CsvTest::testGetApi, with the following error:

```
ErrorException: json_encode(): Invalid UTF-8 sequence in argument

.../Core/Formatters/JSONFormatter.php:56
.../Core/Formatters/JSONFormatter.php:19
.../Core/ContentNegotiator.php:124
...
.../Core/Tests/Api/CsvTest.php:68
```

I tracked this down to the `app/storage/data/tests/csv/latin1.csv` file. I created the following test script to compare the result with an installation where all tests run succesfull.

```
<?php

$handle = fopen("latin.csv", "r");
$text = fread($handle, 10000);
fclose($handle);

echo "Text as read (" . gettype($text) . "):\n";
echo $text;
echo "\n";

$json = json_encode($text);
echo "Text as json (" . gettype($json) . "):\n";
echo $json;
echo "\n";

$json_esc = str_replace("\/", "/", $json);
echo "Text as escaped json (" . gettype($json_esc) . "):\n";
echo $json_esc;
echo "\n\nAll done!\n";
```

Output on a machine where all tests succeed (php 5.5.9):
```
a,b,c
1,2,3
4,5,�
Text as json (boolean):

Text as escaped json (string):


All done!
```

Output on the CentOS machine (php 5.4.16):
```
a,b,c
1,2,3
4,5,�
PHP Warning: json_encode(): Invalid UTF-8 sequence in argument in ...
Text as json (string):
null
Text as escaped json (string):
null

All done!
```

I'm assuming Laravel uses some kind of error handler that converts php warnings to errors, which causes the discrepancy.

**Comment by coreation** [comment]
Hi @DieterDePaepe PHP itself has a warning level where you can configure whether or not certain warnings turn into errors I believe. As they all use the same Laravel version (?) I'm assuming Laravel does not do such a thing. Can you check the php.ini for warning levels?

**Comment by DieterDePaepe** [comment]
I tried with the following values set in php.ini:
```
% PHP Default
error_reporting = E_ALL & ~E_DEPRECATED & ~E_STRICT & ~E_NOTICE
% Recommended for Production
error_reporting = E_ALL & ~E_DEPRECATED & ~E_STRICT
```
And tests still fail.

Laravel does set an error handler (see method register in compiled.php). But the root cause is the difference in PHP versions, because one version causes a PHP warning and one does not. As TDT does not specify a specific php version as requirement, it should work for both versions and the test or affected logic should be investigated.

**Comment by coreation** [comment]
@DieterDePaepe I just checked the [travis-ci configuration](https://travis-ci.org/tdt/core), everything is tested on PHP 5.4. Which makes your outcome even more strange. The exact version is 5.4.37, could you try that version and see if anything changes?

**Comment by DieterDePaepe** [comment]
Output on the same CentOS installation with php 5.4.37 is working without warning:

```
Test as read (string):
a,b,c
1,2,3
4,5,�
Text as json (string):
null
Text as escaped json (string):
null

All done!
```

**Comment by coreation** [comment]
Ok, I'll add a note that it should be the latest 5.4 (37 is very close). If anyone feels like it, they can figure out the difference, perhaps I can look for it but not right now, closing it for now though :).

**Comment by DieterDePaepe** [comment]
I looked through the changelog, I'm guessing it's a fix in 5.4.23.

> Fixed whitespace part of #64874 ("json_decode handles whitespace and case-sensitivity incorrectly").
(http://php.net/ChangeLog-5.php#5.4.23)

**Comment by coreation** [comment]
Aha, that might indeed explain it, so people should actually upgrade their 5.4 install either way, good news for us ;).
